### PR TITLE
Slightly optimizes reagent splashing

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request
Ok so like, before, splashing a reagent performed a rudimentary floodfill based off atmos connectivity.

This was really slow, because it did it using orange(), and repeated orange()s to cut out JUST the rim, because we
needed to ensure things were ordered.

I've changed this to use floodfill. I've also moved some code that was in a second loop into the first one, and replaced a repeated in check with a single use of &

This is still not optimal for large ranges, because we filter by connectivity first and THEN view, but it's faster for smaller ones.

BTW I'm also capping the max spread range at 40x40 tiles. If you want more then 1600 you can rot in hell.

This takes the (uncapped range) cost of deconstructing a highcap tank from 40 seconds to 0.16.

I hate this codebase

## Why It's Good For The Game

🎵Speed annnnd directionnnn🎵.

## Changelog

:cl:
fix: Breaking reagent tanks will be significantly more responsive
/:cl:

### CODE REVIEW

File: code/modules/reagents/chem_splash.dm [hunk 1]
```diff
@@ -75,51 +75,41 @@
...
+ * - spread_range: The range in which to spread the reagents. Will not go over 20
  */
 /proc/spread_reagents(datum/reagents/source, atom/epicenter, spread_range)
+	spread_range = min(spread_range, 20) // Fuck off with trying to do more then this
 	var/datum/effect_system/steam_spread/steam = new /datum/effect_system/steam_spread()
 	steam.set_up(10, 0, epicenter)
 	steam.attach(epicenter)
 	steam.start()
 
+	// This is a basic floodfill algorithem of atmos connected tiles
```
Comment (san7890) [inline code/modules/reagents/chem_splash.dm:87]:
```suggestion
	// This is a basic floodfill algorithm of atmos connected tiles
```

**Review by dragomagol — APPROVED** [review]

File: code/modules/reagents/chem_splash.dm [hunk 2]
```diff
@@ -75,51 +75,41 @@
...
+		if(get_dist(valid_step, epicenter) > spread_range) // We are over threshold, don't add anything new and just keep goin
+			turflist.Cut(i, i+1)
+			i--
+			continue
 
-	var/list/viewable = view(spread_range, epicenter)
-	var/list/accessible = list(epicenter)
-	for(var/i in 1 to spread_range)
-		var/list/turflist = list()
-		for(var/turf/T in (orange(i, epicenter) - orange(i-1, epicenter)))
```
Comment (Kylerace) [inline code/modules/reagents/chem_splash.dm:91]: lol

File: code/modules/reagents/chem_splash.dm [hunk 3]
```diff
@@ -75,51 +75,41 @@
...
-			reactable |= A
-		if(chem_temp >= 300)
-			T.hotspot_expose(chem_temp*2, 5)
-	if(!reactable.len) //Nothing to react with. Probably means we're in nullspace.
-		return
-	for(var/thing in reactable)
-		var/atom/A = thing
-		var/distance = max(1, get_dist(A, epicenter))
+		reactable += valid_step
+		reactable += valid_step.get_all_contents() // Yes this means multitile objects double react. I don't care. skill issue
```
Comment (Kylerace) [inline code/modules/reagents/chem_splash.dm:107]: muh tram

Comment (Kylerace) [reply code/modules/reagents/chem_splash.dm:107]: use |= and get_all_contents() returns src anyways

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chem_splash.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/modules/reagents/chem_splash.dm:107]: if you have a multitile object splashed, it should be splashed by the tiles it's on
also ok

**Pull request merged**